library/std/src/os/fd/owned.rs

```diff
@@ -76,10 +76,18 @@ impl BorrowedFd<'_> {
 }
 
 impl OwnedFd {
-    /// Creates a new `OwnedFd` instance that shares the same underlying file handle
-    /// as the existing `OwnedFd` instance.
-    #[cfg(not(target_arch = "wasm32"))]
+    /// Creates a new `OwnedFd` instance that shares the same underlying file
+    /// description as the existing `OwnedFd` instance.
     pub fn try_clone(&self) -> crate::io::Result<Self> {
+        self.as_fd().try_clone_to_owned()
+    }
+}
+
+impl BorrowedFd<'_> {
+    /// Creates a new `OwnedFd` instance that shares the same underlying file
+    /// description as the existing `BorrowedFd` instance.
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
Does WASI not have a dup function?

Reply:
It does not, today. Because of all the descriptor vs. description issues, the original WASI left `dup` out to give itself flexibility to figure out what it wanted to say. We have a clear sense of how we want this to work, and I expect WASI will add a `dup`. When it does, we can remove this code from Rust.